atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolTest.java

```diff
@@ -110,23 +117,23 @@ public void setup() {
         when(config.getKeyspaceOrThrow()).thenReturn("ks");
         blacklist = new Blacklist(config, Refreshable.only(UNRESPONSIVE_HOST_BACKOFF_SECONDS));
 
-        doAnswer(invocation -> poolServers.add(getInvocationAddress(invocation)))
+        doAnswer(invocation -> poolServers.putIfAbsent(
+                        getCassandraServerFromInvocation(invocation), mock(CassandraClientPoolingContainer.class)))
                 .when(cassandra)
                 .addPool(any());
-        doAnswer(invocation -> poolServers.add(getInvocationAddress(invocation)))
+        doAnswer(invocation -> poolServers.putIfAbsent(
+                        getCassandraServerFromInvocation(invocation), getCassandraContainerFromInvocation(invocation)))
                 .when(cassandra)
-                .returnOrCreatePool(any(), any());
-        doAnswer(invocation -> {
-                    poolServers.remove(getInvocationAddress(invocation));
-                    return mock(CassandraClientPoolingContainer.class);
-                })
+                .addPool(any(), any());
+        doAnswer(invocation -> mock(CassandraClientPoolingContainer.class))
                 .when(cassandra)
-                .removePool(any());
-        doAnswer(invocation -> poolServers.stream()
-                        .collect(Collectors.toMap(x -> x, x -> mock(CassandraClientPoolingContainer.class))))
+                .createPool(any());
+        doAnswer(invocation -> poolServers.remove(getCassandraServerFromInvocation(invocation)))
                 .when(cassandra)
-                .getPools();
+                .removePool(any());
+        doReturn(poolServers).when(cassandra).getPools();
         when(config.socketTimeoutMillis()).thenReturn(1);
+        setupHostsWithInconsistentTopology();
```

Comment:
(a) yup makes sense, overloaded!
(b) removed, although fwiw it shouldn't be retrying at all, since we're mocking setting the return of `getNewHostsWithInconsistentTopologiesAndRetry` -- IIRC it's totally valid to overwrite an object already mocked?

Reply:
For b) Turns out I was wrong.- I thought 
```
when(config.autoRefreshNodes()).thenReturn(true);
when(config.autoRefreshNodes()).thenReturn(false);
```

was identical to `when(config.autoRefreshNodes()).thenReturn(true).thenReturn(false);`
but a test tells me I'm wrong :/